core/iwasm/include/wasm_c_api.h

```diff
@@ -527,6 +527,12 @@ typedef struct WASMModuleCommon *wasm_module_t;
 #define LOAD_ARGS_OPTION_DEFINED
 typedef struct LoadArgs {
     char *name;
+    /* True by default, used by wasm-c-api only.
+    If false, the wasm input buffer (wasm_byte_vec_t) is referenced by the
+    module instead of being cloned. Hence, it can be freed after module loading. */
+    bool clone_wasm_binary;
+    /* This option is only used by the AOT/wasm loader (see wasm_export.h) */
+    bool wasm_binary_freeable;
```

Comment:
i don't understand why this needs this complex.

i guess you just need two modes at most.
a. standard-compatible behavior
b. incompatible, but might save memory

Reply:
We had a discussion here: https://github.com/bytecodealliance/wasm-micro-runtime/pull/3389#discussion_r1592616397

Reply:
> We had a discussion here: [#3389 (comment)](https://github.com/bytecodealliance/wasm-micro-runtime/pull/3389#discussion_r1592616397)

where? this link doesn't seem working for me.